### PR TITLE
feat: add new sidebar

### PR DESCRIPTION
**Pull request by DanielGovrin**
added a new sidebar to the ecommerce shop.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+
+<head>
+  <meta charset="UTF-8" />
+  <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  
+  <title>GeekBazar</title>
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@100&display=swap" rel="stylesheet">
```
Comment (PeterShershov) [inline index.html]: consider using `@import` in the stylesheet

**Review by PeterShershov — COMMENTED** [review]
`npm i react-icons`

**Review by PeterShershov — COMMENTED** [review]
image

File: src/Components/sidebar.module.css [hunk 2]
```diff
@@ -0,0 +1,51 @@
+li{
+    list-style-type: none;
+}
+
+.navbar {
+    background: linear-gradient(to right, #ffffff, #f0f0f0);
+    max-width: 25%;
```
Comment (PeterShershov) [inline src/Components/sidebar.module.css]: no need for that

Comment (PeterShershov) [reply src/Components/sidebar.module.css]: Instead, you can just use flex proportions.

File: src/Components/sidebar.module.css [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+.text {
+    font-size: 1.5rem;
+    padding: 0.5rem 0rem 0rem 0.9rem;
+    font-family: 'Poppins', sans-serif;
+}
+
+.navbar a {
+    color: #01001e;
+    text-decoration: none;
+    transition: color 0.3s ease;
```
Comment (PeterShershov) [inline src/Components/sidebar.module.css]: Please take a look, doesnt do anything.

File: src/Components/sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+type Props = {}
+
+export default function Sidebar({ }: Props) {
+  return (
+    <ul className={styles.navbar}>
+      {SidebarData.map((item, index) => {
+        return (
+          <li key={index} className={styles.listItem}>
+            <Link to={item.path} draggable={false}>
+              <div className={styles.sidebarContent}>
```
Comment (PeterShershov) [inline src/Components/sidebar.tsx]: redundant, styles can be applied on the Link.

File: src/Components/sidebar.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+                <div className={styles.icon}>{item.icon}</div>
```
Comment (PeterShershov) [inline src/Components/sidebar.tsx]: consider removing the wrapping div and target the svg directly using css direct child selector

File: src/app.module.css [hunk 6]
```diff
@@ -0,0 +1,18 @@
+* {
+    margin: 0;
+}
```
Comment (PeterShershov) [inline src/app.module.css]: move to `index.css` and add `padding: 0` 😉

File: package.json [hunk 7]
```diff
@@ -11,7 +11,8 @@
   },
   "dependencies": {
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "react-router-dom": "^6.18.0"
```
Comment (PeterShershov) [inline package.json]: add prettier and prettify your project (incl. stylesheets)

File: index.html (hunk 1 above)
Comment (PeterShershov) [reply index.html]: add linting as well

File: src/Components/SidebarData.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import * as FaIcons from 'react-icons/fa' 
+import * as IoIcons from 'react-icons/io'
+import * as IoIcons5 from 'react-icons/io5'
+import * as PiIcons from 'react-icons/pi'
+
+
```
Comment (PeterShershov) [inline src/Components/SidebarData.tsx]: extra lines

File: src/Components/SidebarData.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+export const SidebarData = [
```
Comment (PeterShershov) [inline src/Components/SidebarData.tsx]: lowercase

File: src/Components/sidebar.module.css [hunk 10]
```diff
@@ -0,0 +1,51 @@
+li{
```
Comment (PeterShershov) [inline src/Components/sidebar.module.css]: index.css

File: src/Components/sidebar.module.css [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    height: 100vh;
+}
+
+.text {
+    font-size: 1.5rem;
+    padding: 0.5rem 0rem 0rem 0.9rem;
+    font-family: 'Poppins', sans-serif;
+}
+
+.navbar a {
```
Comment (PeterShershov) [inline src/Components/sidebar.module.css]: target Link directly with a className

File: src/Components/sidebar.module.css [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+.navbar {
+    background: linear-gradient(to right, #ffffff, #f0f0f0);
+    max-width: 25%;
+    padding: 0.2rem;
+    height: 100vh;
+}
+
+.text {
+    font-size: 1.5rem;
+    padding: 0.5rem 0rem 0rem 0.9rem;
```
Comment (PeterShershov) [inline src/Components/sidebar.module.css]: No need for that. You can align the list item using flex and `align-items: center`.

File: src/Components/sidebar.module.css [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    font-family: 'Poppins', sans-serif;
```
Comment (PeterShershov) [inline src/Components/sidebar.module.css]: consider adding this to index.css as well and applying on all elements.

File: src/Components/sidebar.module.css (hunk 10 above)
Comment (PeterShershov) [reply src/Components/sidebar.module.css]: Please order these selectors by the order of the html elements

File: src/app.module.css [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+.container{
+    display:flex;
+    background-color: #f4f4f4;
+    margin:0;
+}
+
+.sidebar{
```
Comment (PeterShershov) [inline src/app.module.css]: remove

File: src/app.module.css [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+    width:25%;
+    background-color: #333; 
+}
+.content{
+
+    width:100%;
```
Comment (PeterShershov) [inline src/app.module.css]: use flex proportions

File: src/main.tsx [hunk 16]
```diff
@@ -1,10 +1,9 @@
 import React from 'react'
 import ReactDOM from 'react-dom/client'
 import App from './App.tsx'
-import './index.css'
+import { BrowserRouter } from 'react-router-dom'
```
Comment (PeterShershov) [inline src/main.tsx]: remove

File: src/main.tsx [hunk 17]
```diff
@@ -1,10 +1,9 @@
 import React from 'react'
```
Comment (PeterShershov) [inline src/main.tsx]: remove

File: src/Components/sidebar.tsx [hunk 18]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
```
Comment (PeterShershov) [inline src/Components/sidebar.tsx]: remove

File: src/Components/sidebar.tsx [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+import { Link } from 'react-router-dom'
+import styles from './sidebar.module.css'
+import { SidebarData } from './SidebarData'
+type Props = {}
+
+export default function Sidebar({ }: Props) {
```
Comment (PeterShershov) [inline src/Components/sidebar.tsx]: remove props

File: src/Components/sidebar.tsx [hunk 20]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+import styles from './sidebar.module.css'
+import { SidebarData } from './SidebarData'
```
Comment (PeterShershov) [inline src/Components/sidebar.tsx]: rename SidebarData to sideBarData

File: src/Components/sidebar.tsx [hunk 21]
```diff
@@ -0,0 +1,24 @@
...
+type Props = {}
```
Comment (PeterShershov) [inline src/Components/sidebar.tsx]: remove

File: src/App.tsx [hunk 22]
```diff
@@ -1,35 +1,18 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import Sidebar from './components/sidebar'
+import { BrowserRouter} from 'react-router-dom'
+import styles from './app.module.css'
```
Comment (TheAlmightyCrumb) [inline src/App.tsx:3]: Sort imports, you can do so using vscode's commander (Ctrl/Cmd + Shift + P) "Typescript: Sort Imports".

Comment (TheAlmightyCrumb) [inline src/App.tsx]: Sort imports, you can do so using vscode's commander (Ctrl/Cmd + Shift + P) "Typescript: Sort Imports".

File: index.html
Comment (TheAlmightyCrumb) [inline index.html:1]: Add prettier, you can add a configuration file `.prettierrc` to provide specific preferences.
You can also use the [prettier-cli](https://prettier.io/docs/en/cli.html), for example - reformat your entire project using `npx prettier . --write` once you have it installed.

File: src/Components/sidebar.module.css [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+.sidebarContent {
+    display: flex;
+}
+
+.icon {
+    font-size: 1.5rem;
+    padding-top: 0.8rem;
+}
```
Comment (TheAlmightyCrumb) [inline src/Components/sidebar.module.css]: EOF

File: src/Components/sidebar.tsx [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+                <div className={styles.icon}>{item.icon}</div>
+                <span className={styles.text}>{item.title}</span>
+              </div>
+            </Link>
+          </li>
+        )
+      })}
+    </ul>
+  )
+}
```
Comment (TheAlmightyCrumb) [inline src/Components/sidebar.tsx]: EOF

File: src/Components/sidebar.tsx (hunk 20 above)
Comment (DanielGovrin) [reply src/Components/sidebar.tsx]: renamed it to just "data" since it will hold the data for the other pages as well.

File: src/main.tsx [hunk 25]
```diff
@@ -1,10 +1,9 @@
...
-import './index.css'
+import { BrowserRouter } from 'react-router-dom'
+import './app.module.css'
 
 ReactDOM.createRoot(document.getElementById('root')!).render(
-  <React.StrictMode>
     <App />
-  </React.StrictMode>,
-)
+)
```
Comment (TheAlmightyCrumb) [inline src/main.tsx]: here as well

File: src/Components/sidebar.module.css (hunk 3 above)
Comment (DanielGovrin) [reply src/Components/sidebar.module.css]: When I remove it it does change the color, ill be happy to check it again together.

**Review by TheAlmightyCrumb — APPROVED** [review]

**Pull request merged**